### PR TITLE
Changing principle selection component

### PR DESCRIPTION
**Pull request by Shpaackle**
got PrinciplesService to correctly work.  

Need to read more on Observables before I implement them fully.

### CODE REVIEW

File: src/app/shared/services/principles.service.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import principlesData from '../../../assets/tables/principles-data.json';
+import { Observable, of } from 'rxjs';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class PrinciplesService {
+    principles: IPrinciple[];
+
+    private principlesByCategory: { [categoryName: string]: Principle[] } = {};
```
Comment (chriscamac) [inline src/app/shared/services/principles.service.ts:14]: nice job creating an anonymous type for this!

File: src/app/shared/services/principles.service.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import { IPrinciple, Principle } from '../principle.model';
+
+import principlesData from '../../../assets/tables/principles-data.json';
+import { Observable, of } from 'rxjs';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class PrinciplesService {
+    principles: IPrinciple[];
```
Comment (chriscamac) [inline src/app/shared/services/principles.service.ts:12]: this was only for testing right, so we should delete it?

**Review by chriscamac — APPROVED** [review]
overall looks like a great improvement.  Glad you were able to get the service working for you.

File: src/app/shared/services/principles.service.ts (hunk 1 above)
Comment (Shpaackle) [reply src/app/shared/services/principles.service.ts:14]: Thanks, I'm actually proud of figuring out how to do it.  I really like having things hard typed, especially when the IDE warns you that you aren't returning the correct things.

File: src/app/shared/services/principles.service.ts (hunk 2 above)
Comment (Shpaackle) [reply src/app/shared/services/principles.service.ts:12]: Forgot that was only for testing, removing and then going to merge the branch.

**Pull request merged**